### ISSUE TITLE
DialogTrigger: popover variant dismisses unexpectedly with VoiceOver and Chrome

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#4121]
# 🐛 Bug Report 

## 🤔 Expected Behavior

Popover variant of DialogTrigger should not dismiss when attempting to navigate the dialog contents using the VoiceOver cursor.

## 😯 Current Behavior

Popover variant of DialogTrigger dismisses unexpectedly when attempting to navigate the dialog contents using the VoiceOver cursor in Chrome.

Steps to reproduce:

1. Open https://react-spectrum.adobe.com/react-spectrum/DialogTrigger.html#example in Google Chrome with VoiceOver running on macOS.
2. Press Tab key or navigate to set focus to the DialogTrigger Button in the Example reading, "Disk Status."
3. Press Enter or Space to open the popover Dialog.
4. VoiceOver focus should be on the Dialog itself and announce as "dialog with 3 items" followed by instructions on how to interact with the Dialog content.
5. Press `Control`+`Option`+`Shift`+`ArrowDown` to navigate down into the Dialog content.
6. Dialog will close immediately and unexpectedly after the VoiceOver cursor focuses the Dialog heading reading "C://."

Can also reproduce using MS Edge, but can't reproduce using Safari or Firefox.

## 💁 Possible Solution

Don't close `onBlurWithin` if `event.relatedTarget === null` replacing this:
https://github.com/adobe/react-spectrum/blob/d7cb26e27cac7b43f6fdff68ce4c10c1185d1bb3/packages/%40react-aria/overlays/src/useOverlay.ts#L127-L139

with:
```jsx
    onBlurWithin: (e) => {
      // If focus is moving into a child focus scope (e.g. menu inside a dialog),
      // do not close the outer overlay. At this point, the active scope should
      // still be the outer overlay, since blur events run before focus.
      if (
        (e.relatedTarget && isElementInChildOfActiveScope(e.relatedTarget)) ||
        (e.relatedTarget === null && !shouldCloseOnInteractOutside)
      ) {
        return;
      }

      if (!shouldCloseOnInteractOutside || shouldCloseOnInteractOutside(e.relatedTarget as Element)) {
        onClose();
      }
    }
  });
```

## 🔦 Context

Accessibility audit

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | [@adobe/react-spectrum@3.25.0](https://github.com/adobe/react-spectrum/releases/tag/%40adobe%2Freact-spectrum%403.25.0)
| Browser          | Chrome Version 110.0.5481.177 (Official Build) (arm64)
| Operating System | macOS Ventura 13.2.1 (22D68)

## 🧢 Your Company/Team

Adobe/Accessibility

## 🕷 Tracking Issue (optional)

A11Y-5831

**Comment by ktabors** [comment]
The popover no longer closes for me. We believe #4670 fixed this. @majornista please confirm it works now.

**Comment by snowystinger** [comment]
I can't reproduce anymore, we can reopen if we find it's still an issue
